tests/azul_agent.py

```diff
@@ -8,10 +8,15 @@ class AzulAgent:
 
     def __init__(self, deployment):
         self.azul_url = self.AZUL_URL_TEMPLATE.format(deployment=deployment)
+        self.https_session = requests.Session()
 
     def get_specimen_by_project(self, project_shortname):
         filters = {"file": {"project": {"is": [project_shortname]}}}
         path = '/repository/specimens?filters=' + urllib.parse.quote(json.dumps(filters))
-        response = requests.get(self.azul_url + path)
+        from urllib3 import Retry
+        azul_retries = Retry(status_forcelist=(500, 502, 503, 504))
+        azul_adapter = requests.adapters.HTTPAdapter(max_retries=azul_retries)
+        self.https_session.mount('https://', azul_adapter)
```

Comment:
These four lines are once per session, not once per request and should also be in __init__.